client/protocols/collaborate.py

```diff
@@ -0,0 +1,375 @@
+from client.protocols.common import models
+from client.protocols.grading import grade
+from client.utils import output
+from client.utils import auth
+
+import client
+
+import os
+import sys
+import shutil
+
+import json
+import logging
+import urllib.error
+import urllib.request
+import platform
+import time
+
+import webbrowser
+
+log = logging.getLogger(__name__)
+
+class CollaborateProtocol(models.Protocol):
+
+    # Timeouts are specified in seconds.
+    SHORT_TIMEOUT = 30
+    LONG_TIMEOUT = 30
+    API_ENDPOINT = '{prefix}://{server}'
+    FIREBASE_CONFIG = {
+        'apiKey': "AIzaSyAFJn-q5SbxJnJcPVFhjxd25DA5Jusmd74",
+        'authDomain': "ok-server.firebaseapp.com",
+        'databaseURL': "https://ok-server.firebaseio.com",
+        'storageBucket': "ok-server.appspot.com"
```

Comment:
That's the official storage bucket for our project. Firebase gave this auth block to me.